sc/src/main/java/uk/ac/ebi/atlas/search/JsonBioentityInformationController.java

```diff
@@ -79,9 +76,8 @@ public String getBioentityInformation(@PathVariable String geneId) {
         o.addProperty("name", "ExpressionAtlas");
         o.add("values", values);
 
-        //Added a comma',' and end bracket '[' which was removed earlier to maintain valid Json format
-        String result = bioentityProperties + "," + GSON.toJson(o) + "]";
+        jsonArray.add(o);
 
-        return result;
+        return jsonArray.toString();
```

Comment:
Use `GSON.toJson(jsonArray);`

Even if `toString` does what you’d expect it has the following downsides:
1. Maybe the contract is changed, `toString` could just well be the hash of the object.
2. Gson objects can be built with options to do lenient serialization, pretty printing (with indentations) and other options that aren’t possible with `toString`.

Remember: `Gson.fromJson` reads JSON, `Gson.toJson` writes JSON.